ConsentViewController/Assets/GDPRJSReceiver.js

```diff
@@ -42,31 +42,39 @@
   }
 
   var handleMessageEvent = function(SDK) {
-    return function(name, payload) {
-      switch(name) {
+    return function(eventData) {
+      switch(eventData.name) {
         case "sp.showMessage":
-          SDK.onMessageReady();
+          eventData.fromPM ? SDK.onPMReady() : SDK.onMessageReady();
           break;
         case "sp.hideMessage":
-          payload.actionType ?
-            SDK.onAction(getActionFromPM(payload)) :
-            SDK.onAction(getActionFromMessage(payload));
+          eventData.payload.actionType ?
+            SDK.onAction(getActionFromPM(eventData.payload)) :
+            SDK.onAction(getActionFromMessage(eventData.payload));
           break;
         default:
-          payload.action = name;
-          SDK.onMessageEvent(payload);
+          eventData.payload.action = eventData.name;
+          SDK.onMessageEvent(eventData.payload);
       }
     }
   };
 
+  function isFromPM(event) {
+    return event.settings == null
```

Comment:
@andresilveirah in GDPR we have to check **vendorList**attribute and in CCPA we are checking **settings** attribute